src/Java/gtPlusPlus/xmod/gregtech/common/tileentities/machines/multi/production/GregtechMTE_NuclearReactor.java

```diff
@@ -342,155 +356,121 @@ public boolean turnCasingActive(final boolean status) {
 		return true;
 	}
 
+	public FluidStack[] getStoredFluidsAsArray() {	
+		final ArrayList<FluidStack> tFluids = this.getStoredFluids();
+		FluidStack[] aStored = new FluidStack[tFluids.size()];	
+		for (int i = 0; i < aStored.length; i++) {
+			aStored[i] = tFluids.get(i);
+		}
+		return aStored;
+	}
+
+	public int getStoredFuel(GT_Recipe aRecipe) {
+		int aFuelStored = 0;
+		FluidStack aFuelFluid = null;
+		for (FluidStack aFluidInput : aRecipe.mFluidInputs) {
+			if (!aFluidInput.getFluid().equals(NUCLIDE.Li2BeF4.getFluid())) {
+				aFuelFluid = aFluidInput;
+				break;
+			}
+		}
+		if (aFuelFluid != null) {
+			for (GT_MetaTileEntity_Hatch_Input aInputHatch : this.mInputHatches) {
+				if (aInputHatch.getFluid() != null && aInputHatch.getFluidAmount() > 0) {
+					if (aInputHatch.getFluid().isFluidEqual(aFuelFluid)) {
+						aFuelStored += aInputHatch.getFluidAmount();
+					}
+				}
+			}
+		}		
+		return aFuelStored;
+	}
+
 	@Override
 	public boolean checkRecipe(final ItemStack aStack) {
-		final ArrayList<FluidStack> tFluids = this.getStoredFluids();
-		final Collection<GT_Recipe> tRecipeList = GTPP_Recipe.GTPP_Recipe_Map.sLiquidFluorineThoriumReactorRecipes.mRecipeList;
-		if((tFluids.size() > 0) && (tRecipeList != null)) { //Does input hatch have a LFTR fuel?
+		// Warm up for 4~ minutes
+		Logger.WARNING("Checking LFTR recipes.");
```

Comment:
This makes it warm up even with no input. Not sure if this is what you want.

Reply:
It's fine.